javagen/src/main/java/com/azure/autorest/template/ServiceAsyncClientTemplate.java

```diff
@@ -23,17 +23,8 @@ public static ServiceAsyncClientTemplate getInstance() {
     return _instance;
   }
 
-  protected void addAnnotationImports(Set<String> imports) {
-    imports.add("com.azure.core.annotation.ServiceClient");
-  }
-
-  protected void addClassAnnotation(JavaFile javaFile, ServiceClient serviceClient) {
-    javaFile.annotation(String.format("ServiceClient(builder = %s.class, isAsync = true)",
-            serviceClient.getInterfaceName() + ClientModelUtil.getBuilderSuffix()));
-  }
-
   @Override
-  public final void write(AsyncSyncClient asyncClient, JavaFile javaFile) {
```

Comment:
Try to avoid this, IMHO. This class is also used by other generators. We wouldn't know all the implications unless we also test all other generators.

Reply:
discussed this offline, validated that we're not changing the contract here